app/models/adoption_application_pet.rb

```diff
@@ -0,0 +1,21 @@
+class AdoptionApplicationPet < ApplicationRecord
+   belongs_to :adoption_application
+   belongs_to :pet 
+
+   validates :adoption_application_id, presence: true
+   validates :pet_id, presence: true
+   
+   def change_app_pet_status(new_status)
+      self.update_column(:status, new_status)
+   end
+
+   def self.change_status_to_approved(application_id, pet_id)
+      adoption_app_pet = where({pet_id: pet_id,adoption_application_id: application_id}).first
```

Comment:
.where is a good choice here, and another option is `.find_by` which will allow you to pass in both `application_id` and `pet_id` and it will return just one object so you wouldn't have to call `first`. Either .where or `.find_by` are fine to use!